fecgov/fec-cms
# [Don't merge yet!] WIP - 4906 replace typeahead (search)

## Summary

- Resolves #4906 

Replacing Typeahead with autocomplete

### Required reviewers

- couple front-end
- UX
- others to put it through its paces

## Impacted areas of the application

Site search and possibly text-search filters (in case I broke those).
- Header search, site-wide
- [Search page](http://127.0.0.1:8000/search/)

## Screenshots

Should be no visual changes

## Related PRs

None

## Known issues
Non-predictive text searches (e.g. legal search) are not part of the Typeahead / Autosuggest work

I'm getting a browser error: `Uncaught (in promise) Error: The message port closed before a response was received.` but things seem to be working otherwise.

Typeahead filters will be replaced with a different ticket

## How to test

There are a lot of places to test and a lot of ways, but they should generally all pass or fail the same way.

Generally, we want to make sure it
- looks right
  - mobile
  - desktop
- works as expected
  - keyboard typing + enter
  - keyboard typing + clicking an option
  - keyboard typing + arrow/tab navigation
  - keyboard typing + arrow/tab navigation with voiceover turned on

In these locations:
- Header search - check the site header search anywhere on the site
- [Search page](http://127.0.0.1:8000/search/)
- Search page header search (just make sure the header search still works on this page, making sure there are no conflicts with initializing both on one page)
